### PR TITLE
skipperizing the repo

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
skipper is a tool that allows executing all Makefile commands in the container enviroment, without any need to install project dependencies on the host. It is especially useful when working on different multiple projects. All it requires is installing docker and skipper on the host.
skipper is used besides the Makefile, meaning: if somebody does not want to use skipper, he can run makefile commands as usual. skipper can be used on Golang project, Python projects and almost any other programical language
Example:
1) without skipper:
    install GO on your host
    run: make build
         make image
         make deploy
2) with skipper:
   run:
        skipper make build
        skipper make make
        skipper make deploy

skipper also supports passing environment variable to Makefile. skipper repo: https://github.com/Stratoscale/skipper

### CODE REVIEW

File: skipper.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+registry: quay.io
```
Comment (marquiz) [inline skipper.yaml:1]: Does this matter (is it needed). Just `quay.io` looks a bit random 🤔 Could it be just empty string, or `local` or `registry.k8s.io` WDYT?

File: skipper.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+containers:
+  node-feature-discovery-operator-build: Dockerfile.node-feature-discovery-operator-build
+volumes:
+    - $HOME/.cache/go-build:$HOME/.cache/go-build
+    - $HOME/go/pkg:/go/pkg
+    - $HOME/.cache/golangci-lint:$HOME/.cache/golangci-lint
+    - $HOME/.docker:$HOME/.docker
+    - $HOME/.kube/:$HOME/.kube
+    - $HOME/.minikube/:$HOME/.minikube
```
Comment (marquiz) [inline skipper.yaml:12]: Are these needed for something? Just thinking the principle of least privileges/smallest blast radius

**Review by marquiz — COMMENTED** [review]
Thanks @yevgeny-shnaidman for the contribution.

Could we also have a small update on the documentation (`docs/advanced/developer-guide.md`). For example, describe the same things as in the commit message

File: skipper.yaml (hunk 1 above)
Comment (yevgeny-shnaidman) [reply skipper.yaml:1]: the goal of "registry" is to define the registry where the skipper-built image can be pushed. But in practice, nobody ever pushes it, people just build the image once, locally, if needed, and then use it. I will change it to local

File: skipper.yaml (hunk 2 above)
Comment (yevgeny-shnaidman) [reply skipper.yaml:12]: i will remove kube and minikube mappings.If i am not mistaken, then docker mappings are needed

**Comment by yevgeny-shnaidman** [comment]
> Thanks @yevgeny-shnaidman for the contribution.
> 
> Could we also have a small update on the documentation (`docs/advanced/developer-guide.md`). For example, describe the same things as in the commit message

@marquiz added a skipper section to the docs

**Review by marquiz — APPROVED** [review]
Nice improvement @yevgeny-shnaidman

**Comment by marquiz** [comment]
ping @ArangoGutierrez

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**